### ISSUE TITLE
Data preprocessing for 20newsgroups dataset

### ISSUE THREAD

**Issue by marcospiau** [MilaNLProc/contextualized-topic-models#89]
### Description

Hi, guys. First of all, thanks for sharing your work with us. I'd like to reproduce the experiments on [CTM paper](https://aclanthology.org/2021.acl-short.96/) using the `20newsgroups` dataset. I have some questions:
 - are there any additional preprocessing steps other than those reported on paper (*"removing digits, punctuation, stopwords, and infrequent words"*), e.g. removal of email headers, email addresses etc?
 - is there any data partition (e.g. train/validation), or the reported metrics are calculated on full training data?
 - where does the count of 18,173 documents come from? I checked data from both sklearn and the [url in the paper](http://qwone.com/~jason/20Newsgroups/) and they contain 18,846 docs.

Thanks in advance!

**Comment by marcospiau** [comment]
Hi @vinid! Thank you very much for the detailed and quick reply!

**Comment by A11en0** [comment]
hi, Is there a label file? I want to use it for classification.

**Comment by vinid** [comment]
We don't have it here. I think you need to get it from sklearn [datasets](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_20newsgroups.html).

An alternative might be using @silviatti's [OCTIS](https://github.com/MIND-Lab/OCTIS#id8)

**Comment by A11en0** [comment]
Thanks!

**Comment by lei-liu1** [comment]
Hello @vinid! I am replicating the results of your paper and wondering the split of the datasets. 
> reported metrics are computed on the topic lists

And I could not understand this. Could you please explain it specifically?

**Comment by vinid** [comment]
Hello! :) 

there's no split done on the datast.

What I mean with that sentence is that we:

1. run the topic model
2. collected the topics generated by the model
3. computed the various metrics over the topic lists

we did this for 30 times for each model and collected the average values.

Let me know if this makes sense!

**Comment by lei-liu1** [comment]
Thanks a lot for your quick response!
But if there is no split, would the model overfit on the datasets?

**Comment by A11en0** [comment]
> Thanks a lot for your quick response! But if there is no split, would the model overfit the datasets?

Yes, it's a little bit strange. A topic model is different from the normal cluster task. It can be made an inference on the new unseen dataset. I read some papers that split the evaluate type with **transductive** and **inductive,** the former suggests using the full dataset, and later like a classification task i.e. train a model first on training sets and then inference on unseen testing sets. Refer to https://ojs.aaai.org/index.php/AAAI/article/view/6152.
.

**Comment by lei-liu1** [comment]
> Yes, it's a little bit strange. A topic model is different from the normal cluster task. It can be made an inference on the unseen dataset. I read some papers that split the evaluate type with **transductive** and **inductive,** the former suggests using the full dataset, and later like a classification task i.e. train a model first on training sets and then inference on unseen testing sets. Refer to https://ojs.aaai.org/index.php/AAAI/article/view/6152.

Thank you! I will look at the paper.
